sagemaker-spark/pyspark_mnist/pyspark_mnist_custom_estimator.ipynb

```diff
@@ -0,0 +1,342 @@
+    "from sagemaker_pyspark import IAMRole\n",
+    "from sagemaker_pyspark import RandomNamePolicyFactory\n",
+    "from sagemaker_pyspark import EndpointCreationPolicy\n",
+    "\n",
+    "# Use the Docker Registry paths for the K-Means containers from each region\n",
+    "kmeans_ecr_region_path = {\n",
+    "    'us-west-2': '174872318107.dkr.ecr.us-west-2.amazonaws.com',\n",
+    "    'us-east-1': '382416733822.dkr.ecr.us-east-1.amazonaws.com',\n",
+    "    'us-east-2': '404615174143.dkr.ecr.us-east-2.amazonaws.com',\n",
+    "    'eu-west-1': '438346466558.dkr.ecr.eu-west-1.amazonaws.com'\n",
+    "}\n",
+    "\n",
+    "# Create an Estimator from scratch\n",
+    "estimator = SageMakerEstimator(\n",
+    "    trainingImage = kmeans_ecr_region_path[region] + \"/kmeans:1\", # Training image \n",
+    "    modelImage = kmeans_ecr_region_path[region] + \"/kmeans:1\", # Model image\n",
+    "    requestRowSerializer = ProtobufRequestRowSerializer(),\n",
+    "    responseRowDeserializer = KMeansProtobufResponseRowDeserializer(),\n",
+    "    hyperParameters = {\"k\": \"10\", \"feature_dim\": \"784\"}, # Set parameters for K-Means\n",
+    "    sagemakerRole = IAMRole(role),\n",
+    "    trainingInstanceType = \"ml.m4.xlarge\",\n",
+    "    trainingInstanceCount = 1,\n",
+    "    endpointInstanceType = \"ml.t2.medium\",\n",
+    "    endpointInitialInstanceCount = 1,\n",
+    "    trainingSparkDataFormat = \"sagemaker\",\n",
+    "    namePolicyFactory = RandomNamePolicyFactory(\"sparksm-4-\"),\n",
+    "    endpointCreationPolicy = EndpointCreationPolicy.CREATE_ON_TRANSFORM\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The main parts of a `SageMakerEstimator` are:\n",
+    "* `trainingImage`: the Docker Registry path where the training image is hosted - can be a custom Docker image hosting your own model, or one of the [Amazon provided images](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-algo-docker-registry-paths.html)\n",
+    "* `modelImage`: the Docker Registry path where the inference image is used - can be a custom Docker image hosting your own model, or one of the [Amazon provided images](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-algo-docker-registry-paths.html)\n",
+    "* `hyperparameters`: the hyper-parameters of the algorithm being created - the values passed in need to be of type string\n",
+    "\n",
+    "To put this `SageMakerEstimator` back into context, let's look at the below architecture that shows what actually runs on the notebook instance and on SageMaker."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![Custom estimator on SageMaker](img/sagemaker-spark-custom-architecture.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's train this estimator by calling fit on it with the training data."
```

Comment:
Can you call out this one and the following cells that will take a bit of time to run?  Thanks.